src/mozanalysis/metrics/__init__.py

```diff
@@ -97,65 +95,40 @@ def from_expr_for(
                 raise ValueError(
                     f"{self.name}: from_expr contains a dataset template but no value was provided."  # noqa:E501
                 ) from e
-        from_expr = self._from_expr.format(dataset=effective_dataset)
-
-        if (self.experiments_column_type is None) or (experiment_slug is None):
-            return from_expr
-
-        else:
-            return """(
-                SELECT *
-                FROM {from_expr} ds
-                WHERE {slug_expr}
```

Comment:
This part was introducing the main problem of removing clients from the analysis if they unenroll during the experiment